### DISCUSSION TITLE
Specific int8 operators for conv and fully connected?

### DISCUSSION THREAD

**Discussion by mikhilg10** [ggml-org/llama.cpp#10151] [Q&A]
In basic gguf format are  all the weights and activations are being processed as fp16 or there are separate operators for int8 and other formats?. Operators like convolutional , fully connected layer and ffn?? 
What I understood from ggml that quantization is storage purposes only. While inference it'll convert qX to fp16 or 32. Is my understanding wrong??
And 
If I want to add the int8 specific operators how to do that. Can anyone help me???

**Comment by FSSRepo** [comment]
Quantization is mostly used to reduce the size of the weights that need to be multiplied (matrix multiplication `ggml_mul_mat`); the rest of the operations are performed in fp16 or fp32, depending on the case.

**Reply by mikhilg10** [reply]
so the activations are used as it is in fp32 and fp16??
